lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
+      var id = getIdValue(self, item);
+      return id === undefined || id === null;
+    });
+    if (isBulkCreate) {
+      return self.create(data, options, cb);
+    } else {
+      return cb(new Error('updateOrCreate does not support bulk mode or any array input'));
```

Comment:
`... or any array input` - that's not correct, is it? We still support array input when items don't contain ids.